pkg/metadata/metadatapb/rpc.pb.go

```diff
@@ -1,1509 +1,516 @@
-func (m *MetricMetadataResponse) GetResult() isMetricMetadataResponse_Result {
-	if m != nil {
-		return m.Result
+func (x *MetricMetadataRequest) ProtoReflect() protoreflect.Message {
+	mi := &file_metadata_metadatapb_rpc_proto_msgTypes[0]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
 	}
-	return nil
+	return mi.MessageOf(x)
 }
 
-func (m *MetricMetadataResponse) GetMetadata() *MetricMetadata {
-	if x, ok := m.GetResult().(*MetricMetadataResponse_Metadata); ok {
-		return x.Metadata
-	}
-	return nil
+// Deprecated: Use MetricMetadataRequest.ProtoReflect.Descriptor instead.
+func (*MetricMetadataRequest) Descriptor() ([]byte, []int) {
+	return file_metadata_metadatapb_rpc_proto_rawDescGZIP(), []int{0}
 }
 
-func (m *MetricMetadataResponse) GetWarning() string {
-	if x, ok := m.GetResult().(*MetricMetadataResponse_Warning); ok {
-		return x.Warning
+func (x *MetricMetadataRequest) GetMetric() string {
+	if x != nil {
+		return x.Metric
 	}
 	return ""
 }
 
-// XXX_OneofWrappers is for the internal use of the proto package.
-func (*MetricMetadataResponse) XXX_OneofWrappers() []interface{} {
-	return []interface{}{
-		(*MetricMetadataResponse_Metadata)(nil),
-		(*MetricMetadataResponse_Warning)(nil),
+func (x *MetricMetadataRequest) GetLimit() int32 {
+	if x != nil {
+		return x.Limit
 	}
+	return 0
 }
 
-type MetricMetadata struct {
-	Metadata map[string]MetricMetadataEntry `protobuf:"bytes,1,rep,name=metadata,proto3" json:"metadata" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
-}
-
-func (m *MetricMetadata) Reset()         { *m = MetricMetadata{} }
-func (m *MetricMetadata) String() string { return proto.CompactTextString(m) }
-func (*MetricMetadata) ProtoMessage()    {}
-func (*MetricMetadata) Descriptor() ([]byte, []int) {
-	return fileDescriptor_1d9ae5661e0dc3fc, []int{2}
-}
-func (m *MetricMetadata) XXX_Unmarshal(b []byte) error {
-	return m.Unmarshal(b)
-}
-func (m *MetricMetadata) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
-	if deterministic {
-		return xxx_messageInfo_MetricMetadata.Marshal(b, m, deterministic)
-	} else {
-		b = b[:cap(b)]
-		n, err := m.MarshalToSizedBuffer(b)
-		if err != nil {
-			return nil, err
-		}
-		return b[:n], nil
+func (x *MetricMetadataRequest) GetPartialResponseStrategy() storepb.PartialResponseStrategy {
+	if x != nil {
+		return x.PartialResponseStrategy
 	}
-}
-func (m *MetricMetadata) XXX_Merge(src proto.Message) {
-	xxx_messageInfo_MetricMetadata.Merge(m, src)
-}
-func (m *MetricMetadata) XXX_Size() int {
-	return m.Size()
-}
-func (m *MetricMetadata) XXX_DiscardUnknown() {
-	xxx_messageInfo_MetricMetadata.DiscardUnknown(m)
+	return storepb.PartialResponseStrategy(0)
 }
 
-var xxx_messageInfo_MetricMetadata proto.InternalMessageInfo
+type MetricMetadataResponse struct {
+	state         protoimpl.MessageState
```

Comment:
Generated structs now contain types that have locks inside hence we cannot copy them. What do you think should be done with them?